lib/ruby/truffle/truffle/bigdecimal.rb

```diff
@@ -181,6 +189,70 @@ def frac
       self
     end
   end
+
+  def to_s(format = 'E')
+    if finite?
+      spaces       = format.to_i
+      prefix       = case
+                     when self > 0
+                       if [' ', '+'].include? format[0]
+                         format[0]
+                       else
+                         ''
+                       end
+                     when self < 0
+                       '-'
+                     when self == 0
+                       ''
+                     end
```

Comment:
the `=` is the clearest here.
Not sure for `case`/`when` but it might make sense to avoid anything that has visually so many levels of indentation (so `when` would have the same indent as the line before).